stdlib/LinearAlgebra/src/givens.jl

```diff
@@ -250,20 +250,23 @@ function givensAlgorithm(f::Complex{T}, g::Complex{T}) where T<:AbstractFloat
     return cs, sn, r
 end
 
-# enable for unitful quantities
-function givensAlgorithm(f::T, g::T) where T
-    fs = f / oneunit(T)
-    gs = g / oneunit(T)
-    typeof(fs) === T && typeof(gs) === T &&
-    !isa(fs, Union{AbstractFloat,Complex{<:AbstractFloat}}) &&
-    throw(MethodError(givensAlgorithm, (fs, gs)))
-
-    c, s, r = givensAlgorithm(fs, gs)
-    return c, s, r * oneunit(T)
+# From Janovská, D., & Opfer, G. (2003). Givens’ Transformation Applied to Quaternion
+# Valued Vectors. BIT Numerical Mathematics, 43(5), 991–1002.
+# doi:10.1023/b:bitn.0000014561.58141.2c
+function givensAlgorithm(f::Number, g::Number)
+    nrm = hypot(f, g)
+    c = abs(f) / nrm
+    s, u = if iszero(f)
+        -one(first(promote(f, g))), -g
+    else
+        # Note that the paper conjugates the argument in the definition of sign but the
+        # givens rotation implementation then ends up with conjugating twice.
+        signf̄ = f / abs(f)
```

Comment:
Could this be replaced by just
```suggestion
        signf̄ = sign(f)
```
given that the fallback definition of `sign(x::Number)` returns `x / abs(x)`? https://github.com/JuliaLang/julia/blob/c0611e8c49c8e279fd83ed57a395a3217ba1dc0c/base/number.jl#L161

Reply:
Good idea. Then you'd benefit from a cheap `sign` whenever it's available. I've made that change